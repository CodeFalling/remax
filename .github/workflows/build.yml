name: Node CI

on: [push]

jobs:
  build:

    runs-on: ${{ matrix.os }}

    strategy:
      matrix:
        node-version: [10.x, 12.x]
        os: [ubuntu-latest, windows-latest]
      fail-fast: false

    steps:
    - uses: actions/checkout@v1
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}
    - name: install
      run: yarn install
    - name: lint
      run: yarn lint --fix
    - name: build
      run: yarn build
    - name: test
      run: yarn test:cov
      env:
        CI: true
    - name: Upload coverage to Codecov
      if: success() && runner.os == 'Linux'
      run: curl -s https://codecov.io/bash | bash -s -- -t $CODECOV_TOKEN
      env:
        CODECOV_TOKEN: ${{secrets.CODECOV_TOKEN}}
