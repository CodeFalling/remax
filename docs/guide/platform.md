---
title: 平台
order: 28
---

Remax 目前支持的平台

- 微信小程序: wechat
- 支付宝小程序: alipay

## 获取平台信息

```js
import { Platform } from 'remax';

console.log(Platform.current) // 当前环境，如 wechat，alipay 或其他
console.log(Platform.isWechat) // 是否是微信
console.log(Platform.isAlipay) // 是否是支付宝

...
```

## 针对平台开发

Remax 为每个小程序平台提供单独的模块，如：

```js
// 微信小程序开发
import { View, navigateTo, useShow } from 'remax/wechat';
```

```js
// 支付宝小程序开发
import { View, navigateTo, useShow } from 'remax/alipay';
```

## 跨平台开发

如果你有跨平台开发的需求，请查阅 [高级指南 - 跨平台开发](/advanced-guide/cross-platform)
