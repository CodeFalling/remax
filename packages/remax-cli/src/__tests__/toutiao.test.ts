import runTest from './helpers/runTest';
import { Platform } from '../build/platform';

describe('build toutiao app', () => {
  runTest('toutiao', Platform.toutiao);
});
