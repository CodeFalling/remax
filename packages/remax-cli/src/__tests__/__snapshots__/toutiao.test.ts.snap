// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toutiao build simple app 1`] = `
Array [
  Object {
    "code": "'use strict';

App({});
",
    "fileName": "__tests__/fixtures/simple/src/app.js",
  },
  Object {
    "code": "'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index$2 = require('../../../../../../npm/react/index.js');
require('../../../../../../npm/remax/lib/render.js');
require('../../../../../../npm/remax/lib/createAppConfig.js');
var createPageConfig = require('../../../../../../npm/remax/lib/createPageConfig.js');
require('../../../../../../npm/remax/lib/index.js');
var View = require('../../../../../../npm/remax/lib/adapters/toutiao/components/View.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Input.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Textarea.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Video.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Swiper.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/ScrollView.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/SwiperItem.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Icon.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Text.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/RichText.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Progress.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Button.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/CheckboxGroup.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Checkbox.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Form.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Label.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Picker.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/PickerView.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/PickerViewColumn.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/RadioGroup.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Radio.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Slider.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Switch.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Navigator.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Image.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/Canvas.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/components/WebView.js');
require('../../../../../../npm/remax/lib/adapters/toutiao/api.js');

var _page = function _page() {
  return index$2.createElement(View.default, null, \\"foo\\");
};

var index = Page(createPageConfig.default(_page));

exports.default = index;
",
    "fileName": "__tests__/fixtures/simple/src/pages/toutiao/index.js",
  },
  Object {
    "code": "<template name=\\"REMAX_TPL\\">
  <block tt:for=\\"{{tree.root.children}}\\" tt:key=\\"{{id}}\\">
    <template is=\\"{{'REMAX_TPL_' + item.type}}\\" data=\\"{{i: item}}\\" />
  </block>
</template>


  

<template name=\\"REMAX_TPL_view\\">
  <view >
    
      <block tt:for=\\"{{i.children}}\\" tt:key=\\"{{id}}\\">
        <template is=\\"{{'REMAX_TPL_' + item.type}}\\" data=\\"{{i: item}}\\" />
      </block>
    
  </view>
</template>


<template name=\\"REMAX_TPL_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
",
    "fileName": "base.ttml",
  },
  Object {
    "code": "{
  \\"pages\\": [
    \\"pages/toutiao/index\\"
  ],
  \\"window\\": {
    \\"navigationBarTitleText\\": \\"Toutiao Todo App\\",
    \\"navigationBarBackgroundColor\\": \\"#323239\\"
  }
}",
    "fileName": "app.json",
  },
  Object {
    "code": "<import src=\\"../../../../../../base.ttml\\"/>
<template is=\\"REMAX_TPL\\" data=\\"{{tree: action}}\\" />
",
    "fileName": "__tests__/fixtures/simple/src/pages/toutiao/index.ttml",
  },
]
`;
